### DISCUSSION TITLE
React Plugin

### DISCUSSION THREAD

**Discussion by meaudotme** [doulevo/doulevo#17] [Ideas]
I'm currently building the React plugin and it's going good enough so far. It's located there - https://github.com/R4meau/doulevo-react-plugin

I wanted to allow users to pick between a TypeScript and a JavaScript template. Problem is, Doulevo doesn't give the possibility to generate the templates at runtime using some command, so I went ahead and tried generating them both pre-build but Doulevo doesn't give the possibility to do a cleanup after the questionnaire, so the resulting project still contains both the TS and the JS folders.

So for now, I'll keep it simple and only provide the TS version. I'll wait for Ashley on here so we can figure this out.

P.S.: I originally posted this on a Discord server I created for this project. It's located there - https://discord.gg/5BRGvMCe

**Comment by ashleydavis** [comment]
Thanks @R4meau, later Doulevo needs functionality to allow plugins to dynamically generate any files they need to, it's just a bit too soon so the functionality isn't there yet.

**Comment by ashleydavis** [comment]
Have you created the repo yet? I can't see it.

**Reply by meaudotme** [reply]
Yes, it's available over there - https://github.com/R4meau/doulevo-react-plugin

**Reply by meaudotme** [reply]
Oops, didn't realize it was private. You can now access it.

**Comment by ashleydavis** [comment]
Thanks, I'll take a look a bit later.

**Comment by ashleydavis** [comment]
Wow, really cool!

Caddy looks really cool, but what if I want to use something else? Or deploy the static web site to something like AWS? 

I've opened a new issue so we can discuss this: https://github.com/doulevo/doulevo/issues/18

**Reply by meaudotme** [reply]
Great. Following up over there.
